### PR TITLE
z/TPF: Add z/TPF as a recognized operating system

### PR DESCRIPTION
**Pull request by jjohnst-us**
Builds for z/TPF are done via cross-compilation.

When using the BOOT_JDK (s390x linux) the JLINK tool validates the OS it's building for.

The proposed update simply adds z/TPF as a recognized Operating System for Semeru.

If accepted looking to get this into JDK 21 Releases/Refreshes.  Please let me know what the earliest release this would be available in, and if it's possible to get the update into in-progress refreshes for Semeru 21.

### CODE REVIEW

**Comment by pshipton** [comment]
There is about 3 weeks to get it into 24_04.

**Review by keithc-ca — CHANGES_REQUESTED** [review]
Were you planning to update `platform.m4` to recognize z/TPF at configuration time?

File: src/java.base/share/classes/jdk/internal/util/OperatingSystem.java [hunk 1]
```diff
@@ -135,6 +139,14 @@ public static boolean isZOS() {
         return PlatformProps.TARGET_OS_IS_ZOS;
     }
 
+    /**
+     * {@return {@code true} if built for the ZTPF operating system}
```
[the remark this answers is not in the document]
Comment (jjohnst-us) [reply src/java.base/share/classes/jdk/internal/util/OperatingSystem.java]: @keithc-ca 
Do I need to add the IBM specific copyright to these segments?

File: make/autoconf/platform.m4 [hunk 2]
```diff
@@ -226,6 +226,10 @@ AC_DEFUN([PLATFORM_EXTRACT_VARS_FROM_OS],
       VAR_OS=aix
       VAR_OS_TYPE=unix
       ;;
+    *ztpf*)
+      VAR_OS=ztpf
+      VAR_OS_TYPE=unix
+      ;;
```
Comment (keithc-ca) [inline make/autoconf/platform.m4:232]: Unfortunately, this would only represent the tip of this particular iceberg. I expect many more changes would be required for proper configuration of builds for z/TPF. Please revert this change.

Comment (jjohnst-us) [reply make/autoconf/platform.m4]: Update to platform.m4 reverted.

**Comment by keithc-ca** [comment]
Jenkins line endings check

**Comment by keithc-ca** [comment]
Jenkins compile amac jdknext

**Review by keithc-ca — APPROVED** [review]
